src/RA_Dlg_MemBookmark.cpp

```diff
@@ -20,437 +22,434 @@ namespace ra {
 {
-    switch (nMsg)
+    const auto OnDestroy = []([[maybe_unused]] HWND) noexcept { g_hIPEEditBM = nullptr; };
+
+    // hwndNewFocus -> wParam
+    const auto OnKillFocus = [](HWND hwnd, [[maybe_unused]] HWND /*hwndNewFocus*/)
     {
-        case WM_DESTROY:
-            g_hIPEEditBM = nullptr;
-            break;
+        NMHDR hdr{hwnd, gsl::narrow<UINT_PTR>(GetDlgCtrlID(hwnd)), LVN_ENDLABELEDIT};
+
+        LVITEM item{};
+        item.mask = LVIF_TEXT;
+        item.iItem = nSelItemBM;
+        item.iSubItem = nSelSubItemBM;
 
-        case WM_KILLFOCUS:
+        LV_DISPINFO lvDispinfo{hdr, item};
         {
-            LV_DISPINFO lvDispinfo;
-            ZeroMemory(&lvDispinfo, sizeof(LV_DISPINFO));
-            lvDispinfo.hdr.hwndFrom = hwnd;
-            lvDispinfo.hdr.idFrom = GetDlgCtrlID(hwnd);
-            lvDispinfo.hdr.code = LVN_ENDLABELEDIT;
-            lvDispinfo.item.mask = LVIF_TEXT;
-            lvDispinfo.item.iItem = nSelItemBM;
-            lvDispinfo.item.iSubItem = nSelSubItemBM;
-            lvDispinfo.item.pszText = nullptr;
-
-            wchar_t sEditText[256];
-            GetWindowTextW(hwnd, sEditText, 256);
-            g_MemBookmarkDialog.Bookmarks()[nSelItemBM]->SetDescription(sEditText);
-
-            HWND hList = GetDlgItem(g_MemBookmarkDialog.GetHWND(), IDC_RA_LBX_ADDRESSES);
-
-            //	the LV ID and the LVs Parent window's HWND
-            SendMessage(GetParent(hList), WM_NOTIFY, static_cast<WPARAM>(IDC_RA_LBX_ADDRESSES), reinterpret_cast<LPARAM>(&lvDispinfo));	//	##reinterpret? ##SD
-
-            DestroyWindow(hwnd);
-            return 0;
+            std::wstring sEditText(256, wchar_t());
+            GetWindowTextW(hwnd, sEditText.data(), 256);
+            g_MemBookmarkDialog.Bookmarks().at(nSelItemBM).SetDescription(sEditText);
         }
 
-        case WM_KEYDOWN:
-        {
-            if (wParam == VK_RETURN || wParam == VK_ESCAPE)
-            {
-                DestroyWindow(hwnd);	//	Causing a killfocus :)
-            }
-            else
-            {
-                //	Ignore keystroke, or pass it into the edit box!
-                break;
-            }
+        HWND hList = GetDlgItem(g_MemBookmarkDialog.GetHWND(), IDC_RA_LBX_ADDRESSES);
 
-            break;
-        }
+        // the LV ID and the LVs Parent window's HWND
+        FORWARD_WM_NOTIFY(GetParent(hList), IDC_RA_LBX_ADDRESSES, &lvDispinfo, ::SendMessage);
+
+        DestroyWindow(hwnd);
+    };
+
+    // vk -> wParam
+    const auto OnKeyDown = [](HWND hwnd, UINT vk, [[maybe_unused]] BOOL, [[maybe_unused]] int,
+                              [[maybe_unused]] UINT) noexcept
+    {
+        if (vk == VK_RETURN || vk == VK_ESCAPE)
+            DestroyWindow(hwnd); // Causing a killfocus :)
+    };
+
+    switch (nMsg)
+    {
+        HANDLE_MSG(hwnd, WM_DESTROY, OnDestroy);
+        HANDLE_MSG(hwnd, WM_KILLFOCUS, OnKillFocus);
+        HANDLE_MSG(hwnd, WM_KEYDOWN, OnKeyDown);
```

Comment:
These changes contradict the changes made in DialogBase and it's children and will be replaced by that logic when this dialog is converted. Also these changes are pretty drastic - I hope you tested them thoroughly.